### PR TITLE
handle deletion gracefully for already terminated master instances

### PR DESCRIPTION
**Pull request by xh3b4sd**

### CODE REVIEW

File: service/controller/v25/resource/tccp/error.go [hunk 1]
```diff
@@ -30,6 +31,28 @@ func IsAlreadyExists(err error) bool {
...
+	Kind: "alreadyTerminatedError",
+}
+
+// IsAlreadyTerminated asserts alreadyTerminatedError.
+func IsAlreadyTerminated(err error) bool {
+	c := microerror.Cause(err)
+
+	aerr, ok := c.(awserr.Error)
+	if ok {
+		if aerr.Code() == "IncorrectInstanceState" {
```
Comment (xh3b4sd) [inline service/controller/v25/resource/tccp/error.go]: > {"caller":"github.com/giantswarm/aws-operator/vendor/github.com/giantswarm/operatorkit/controller/resource/retryresource/resource_wrapper.go:98","controller":"aws-operator","event":"delete","level":"warning","loop":"19","message":"retrying due to error","object":"/apis/provider.giantswarm.io/v1alpha1/namespaces/default/awsconfigs/38z4i","resource":"tccpv25","stack":"[{/go/src/github.com/giantswarm/aws-operator/vendor/github.com/giantswarm/operatorkit/controller/resource/retryresource/resource_wrapper.go:91: } {/go/src/github.com/giantswarm/aws-operator/service/controller/v25/resource/tccp/delete.go:28: } {/go/src/github.com/giantswarm/aws-operator/service/controller/v25/resource/tccp/resource.go:188: } {IncorrectInstanceState: The instance 'i-0b26c88f3546aefee' must be in a 'running', 'pending', 'stopping' or 'stopped' state for this operation.\n\tstatus code: 400, request id: 9d57b148-da42-49bc-ae7f-c121f7c9e38d}]","time":"2019-03-22T14:03:18.216824+00:00","underlyingResource":"tccpv25","version":"109386904"}

**Comment by xh3b4sd** [comment]
Tested on `ginger`. Merging since this fixes an issue with the deletion.

File: service/controller/v25/resource/tccp/error.go (hunk 1 above)
Comment (kopiczko) [reply service/controller/v25/resource/tccp/error.go]: How about adding this as a comment?

```
	IncorrectInstanceState: The instance 'i-0b26c88f3546aefee' must be in a 'running', 'pending', 'stopping' or 'stopped' state for this operation.
```

Comment (xh3b4sd) [reply service/controller/v25/resource/tccp/error.go]: See https://github.com/giantswarm/aws-operator/pull/1524.

**Pull request merged**